src/test/java/com/tabnine/binary/fetch/LocalBinaryVersionsTests.kt

```diff
@@ -0,0 +1,126 @@
+package com.tabnine.binary.fetch
+
+import com.tabnine.general.StaticConfig.* // ktlint-disable no-wildcard-imports
+import com.tabnine.testutils.TestData.* // ktlint-disable no-wildcard-imports
+import io.mockk.every
+import io.mockk.impl.annotations.InjectMockKs
+import io.mockk.impl.annotations.MockK
+import io.mockk.junit5.MockKExtension
+import org.hamcrest.Matchers.empty
+import org.hamcrest.Matchers.equalTo
+import org.junit.Assert.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.Optional
+import java.util.stream.Collectors.toList
+import java.util.stream.Stream
+
+@ExtendWith(MockKExtension::class)
+class LocalBinaryVersionsTests {
+    @TempDir
+    @JvmField
+    var temporaryFolder: Path? = null
+
+    @MockK
+    lateinit var binaryValidator: BinaryValidator
+
+    @InjectMockKs
+    lateinit var localBinaryVersions: LocalBinaryVersions
+
+    private val originalHome = System.getProperty(USER_HOME_PATH_PROPERTY)
+
+    @BeforeEach
+    fun setup() {
+        System.setProperty(USER_HOME_PATH_PROPERTY, temporaryFolder.toString())
+    }
+
+    @AfterEach
+    fun teardown() {
+        System.setProperty(USER_HOME_PATH_PROPERTY, originalHome)
```

Comment:
Shouldn't we also delete the temp folder here?

Reply:
JUnit takes care of that https://junit.org/junit5/docs/5.4.0/api/org/junit/jupiter/api/io/TempDir.html

Reply:
Cool.